processing/src/main/java/io/druid/query/metadata/SegmentAnalyzer.java

```diff
@@ -152,6 +152,6 @@ public ColumnAnalysis analyzeComplexColumn(Column column)
       size += inputSizeFn.apply(complexColumn.getRowValue(i));
     }
 
-    return new ColumnAnalysis(capabilities.getType().name(), size, null, null);
+    return new ColumnAnalysis(typeName, size, null, null);
```

Comment:
yes this is required so that it says "hyperUnique" instead of "COMPLEX"  in case of complex aggregations. that is what the issue being solved really wants.